src/sentry/api/base.py

```diff
@@ -213,14 +218,28 @@ def permission_denied(self, request, message=None, code=None):
         the appropriate exception according to parent DRF function.
         """
         permissions = self.get_permissions()
+        if request.user.is_authenticated and len(permissions) == 1:
```

Comment:
Nice guard check! This makes the code intentions better to read/understand, and helps prevent unnecessary errors!

Reply:
We don't have to block on this PR, but do you think there's a way we can create an easily extendable class for these checks? My thinking here is we've already added 2 different "checks/responses", and tomorrow if we needed to add more, how can I do that without further increasing this if statement so much. It looks like our main components for making this decision are the `permission_cls` and `user` objects. Here's just a suggestion:

```
class PermissionEvaluator:
     def __init__(self, permission_cls: Object, user: User) -> None:
          self._permission_class: Object = permission_cls
          self._user: User = user
          
          self._checks: Dict[Callable[], Exception] = {
                self._super_user_check, SuperuserRequired,
                ...
          }
    
    def _super_user_check(self) -> bool:
           use_staff = features.has("auth:enterprise-staff-cookie", actor=self.user)
           has_only_superuser_permission = isinstance(
                self._permission_class, (SuperuserPermission, SuperuserOrStaffFeatureFlaggedPermission)
            )
           
           return (not use_staff and can_be_superuser and has_only_superuser_permission)
    
    def get_permission_error(self) -> Optional[Exception]:
         for checker, error in self._checks:
               if not checker():
                    continue
              
              return error()
                    
         return None
```

This way the next person knows to just add their implementation and specific error in the evaluator and know it will properly get invoked. Then the code looks like:
```
if request.user.is_authenticated and len(permissions) == 1:
    permission_cls = permissions[0]
    checker = PermissionEvaluator(permission_cls, request.user)
    error = checker.get_permission_error()
    if error is not None:
        raise Error
      
```

Reply:
That looks pretty good but as far I could tell when digging into permissions, I think we only do this for superuser and staff right now. If I find other exceptions being raised I'll come back to this 👍